### PR TITLE
chore: migrate to react-native 0.70.6

### PR DESCRIPTION
**Pull request by ice-hector**

### CODE REVIEW

File: ios/ice.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -782,7 +786,7 @@
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
 				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
-				CLANG_CXX_LANGUAGE_STANDARD = "c++17";
+				CLANG_CXX_LANGUAGE_STANDARD = "c++14";
```
Comment (ice-orion) [inline ios/ice.xcodeproj/project.pbxproj:789]: Why do we downgrade it? There is no such a change in the helper

File: ios/ice.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -844,6 +848,7 @@
 					"-DFOLLY_MOBILE=1",
 					"-DFOLLY_USE_LIBCPP=1",
 				);
+				REACT_NATIVE_PATH = "${PODS_ROOT}/../../node_modules/react-native";
```
Comment (ice-orion) [inline ios/ice.xcodeproj/project.pbxproj:851]: And this change as well lack in the changes, where did it come from?

Comment (ice-hector) [reply ios/ice.xcodeproj/project.pbxproj:855]: Maybe it's autogenerated somehow by the new Xcode version... I've deleted, looks like it works without this line

File: Gemfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-ruby '2.7.4'
+ruby '2.7.5'
```
Comment (ice-ares) [inline Gemfile:3]: Lets try 3.1.3

File: src/api/client/index.ts [hunk 4]
```diff
@@ -105,7 +105,7 @@ export const isApiError = (
   error: unknown,
   expectedStatus?: number,
   expectedCode?: string,
-): error is AxiosError => {
+): error is AxiosError<{code?: string; data?: {[key: string]: unknown}}> => {
```
Comment (ice-ares) [inline src/api/client/index.ts:108]: And message

File: ios/ice.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -92,6 +94,7 @@
 			isa = PBXFrameworksBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				7C44B262292E63A7006C7F5D /* libreact-native-config.a in Frameworks */,
```
Comment (ice-orion) [inline ios/ice.xcodeproj/project.pbxproj:97]: Did you manually link `react-native-config` lib? Or what is this?

File: ios/ice.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -907,6 +915,7 @@
 					"-DFOLLY_MOBILE=1",
 					"-DFOLLY_USE_LIBCPP=1",
 				);
+				REACT_NATIVE_PATH = "${PODS_ROOT}/../../node_modules/react-native";
```
Comment (ice-orion) [inline ios/ice.xcodeproj/project.pbxproj:918]: Do we need it?

File: src/screens/InviteFlow/InviteShare/components/ShareCard/index.tsx [hunk 7]
```diff
@@ -139,6 +140,7 @@ const ShareCard = () => {
             'https://e7.pngegg.com/pngimages/223/378/png-clipart-three-ice-cubes-three-ice-cubes-ice.png',
```
Comment (ice-orion) [inline src/screens/InviteFlow/InviteShare/components/ShareCard/index.tsx:140]: What is it tho? What image should be there?

File: Gemfile (hunk 3 above)
Comment (ice-orion) [reply Gemfile:3]: the ruby version should be also changed in `CICD.yaml`

Comment (ice-ares) [reply Gemfile:3]: lets test and see

File: src/screens/InviteFlow/InviteShare/components/ShareCard/index.tsx (hunk 7 above)
Comment (ice-hector) [reply src/screens/InviteFlow/InviteShare/components/ShareCard/index.tsx:140]: Its hardcoded for Instagram sharing (to stories). Should be replaced with some other in release

File: ios/ice.xcodeproj/project.pbxproj (hunk 6 above)
Comment (ice-hector) [reply ios/ice.xcodeproj/project.pbxproj:918]: Lets delete but as I said before looks like its autogenerated, I didnt add this

File: ios/ice.xcodeproj/project.pbxproj (hunk 5 above)
Comment (ice-hector) [reply ios/ice.xcodeproj/project.pbxproj:97]: Yep, without this, we have compilation issues. I get this solution from issues in their GitHub

File: src/api/client/index.ts (hunk 4 above)
Comment (ice-hector) [reply src/api/client/index.ts:108]: @ice-ares you mean `error` ?

File: ios/ice.xcodeproj/project.pbxproj (hunk 5 above)
Comment (ice-orion) [reply ios/ice.xcodeproj/project.pbxproj:97]: May be then it makes sense to update react-native-config lib itself to fix the compilation issue? They prohibited `react-native link` and raise a warning if it's used -> they state that auto-linking should be used wherever possible

Comment (ice-hector) [reply ios/ice.xcodeproj/project.pbxproj:97]: That is the result of the update already. Now it's 1.4.11 as the latest release
https://github.com/luggit/react-native-config

File: Gemfile (hunk 3 above)
Comment (ice-hector) [reply Gemfile:3]: Fixed version in CICD.yaml, all good now

File: ios/ice.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -135,6 +136,7 @@
 		2D16E6871FA4F8E400B85C8A /* Frameworks */ = {
 			isa = PBXGroup;
 			children = (
+				7C44B261292E63A7006C7F5D /* libreact-native-config.a */,
```
Comment (ice-orion) [inline ios/ice.xcodeproj/project.pbxproj:139]: Do we need it? Turned out manual react-native-config linking is not needed, is it a part of it?

Comment (ice-hector) [reply ios/ice.xcodeproj/project.pbxproj:139]: Removed

**Review by ice-ares — APPROVED** [review]

**Review by ice-orion — APPROVED** [review]

**Pull request merged**